packages/apps/tools/src/services/transfer-tracker/get-transfer-status.ts

```diff
@@ -156,26 +159,39 @@ export async function getXChainTransferInfo({
   senderChain,
   receiverChain,
   network,
+  networksData,
   t,
 }: {
   requestKey: string;
   senderAccount: string;
   senderChain: ChainwebChainId;
   receiverChain: ChainwebChainId;
   network: Network;
+  networksData: INetworkData[];
   t: Translate;
 }): Promise<IStatusData> {
   debug(getXChainTransferInfo.name);
   try {
-    const networkId = chainNetwork[network].network;
+    // const { selectedNetwork: network, networksData } = useWalletConnectClient();
+    const networkData: INetworkData | undefined = networksData.find(
+      (item) => (network as Network) === item.networkId,
+    );
+
+    if (!networkData) return { error: 'No network found' };
 
     const requestObject = {
       requestKey,
-      networkId,
+      networkId: network,
```

Comment:
On other places `networkData.networkId` is used, should we do that here as well, to keep it consistent? 

Also, we're doing this network logic now in multiple places (inside several `services`), should we refactor it to a reusable function perhaps?

Reply:
We already have a `getApiHost` function and `client` to create different clients (for sender and receiver networks and chains). Other than these two reusable functions, what else do you think can be reusable? Generating the request object? Yet again we would need to add all these values as parameters. 
Let me know what do you think

Reply:
Ah right, yeah let's just leave as is then for now 👌